localstack-core/localstack/services/apigateway/patches.py

```diff
@@ -145,6 +145,7 @@ def apigateway_models_stage_to_json(fn, self):
 
         return result
 
+    # TODO remove this patch when the behavior is implemented in moto
```

Comment:
Quick question as this has been an issue in the past, we should try to prevent `-` in custom ids, because it messes up with the regex matching the id (mixing it with the `-vpc` suffix or something), and also it should probably be lower case. Where could we implement this?

Reply:
Thanks for the feedback @bentsku. This is not something that is currently possible. I think it could be done by adding a `validator` method to `ResourceIdentifier`. By default, in would always return True, but then it can be overriden by specific Indentifier. All it would need then is to call it in [find_id_from_sources](https://github.com/getmoto/moto/pull/8216/files#diff-6b1e6c5ba61dca3f691bf713f7fd6715e50f2f815095a57e376dffe6717caf60R111-R125).

It might be easier to implement in moto to prevent patching.

Reply:
Here's how it could look like :thinking: 
https://github.com/localstack/localstack/compare/replicator-user-defined-id-generator...custom-id-validator?expand=1

Reply:
Makes sense, that could be nice! The validator could also log a message, nice that it falls back to the default. We can discuss this, it's currently not the case but sometimes leads to confused users as to why trying to hit their API doesn't match when there's an hyphen.. so thank you for looking into this already!

Reply:
Yah, I think it could be a nice to have. The logging could be improved I agree. I will keep thinking on a nice solution for this as well, and see if I can come up with a more general solution as I see more use cases.